src/macros.rs

```diff
@@ -512,23 +510,21 @@ macro_rules! tarpc_service_then__ {
                   -> ::std::result::Result<$out, $crate::Error<$error>> {
             match msg__ {
                 ::std::result::Result::Ok(msg__) => {
-                    if let Response__::$fn_name(msg__) =
-                        msg__
-                    {
-                        ::std::result::Result::Ok(msg__)
-                    } else {
-                       unreachable!()
+                    #[allow(unreachable_patterns)]
+                    match msg__ {
+                        Response__::$fn_name(msg__) =>
+                            ::std::result::Result::Ok(msg__),
+                        _ => unreachable!(),
```

Comment:
Can this code path be hit by an adversarial client sending bad requests to the server? It would cause the server to crash, right? Or maybe that would just cause a deserialization error?

I'm approving anyways because it looks like this these `unreachable`s were there before.